dht.go

```diff
@@ -507,3 +578,46 @@ func (dht *IpfsDHT) newContextWithLocalTags(ctx context.Context, extraTags ...ta
 	) // ignoring error as it is unrelated to the actual function of this code.
 	return ctx
 }
+
+func (dht *IpfsDHT) handleProtocolChanges(ctx context.Context) {
+	// register for event bus protocol ID changes
+	sub, err := dht.host.EventBus().Subscribe(new(event.EvtPeerProtocolsUpdated))
+	if err != nil {
+		panic(err)
```

Comment:
panic? What if we're shutting down?

Reply:
I suspect the idea when this was put here was that Subscribe should only error if there's a problem creating the subscription which likely means something unrecoverable has happened (developer error passing the wrong type into the eventbus, a bad option, etc.).

If we make Subscription occur in the DHT creation function instead of a separate goroutine then we can return err instead of panicing. If that's of interest comment in https://github.com/libp2p/go-libp2p-kad-dht/pull/472#discussion_r387432720

Reply:
Ah, I see. You're right. Panicing looks like the right thing to do in this case. It looks like it can only happen on programmer error.